### PR TITLE
CI Workflow: Add all Azure supported k8s versions

### PR DESCRIPTION
**Pull request by brlbil**
The current aks conformance tests run against one version of k8s either a specific version or the default version that is provided by the cloud provider.

This PR adds all the supported k8s versions by Azure in a matrix strategy.

Other cloud providers will be added by subsequent PRs.

Normal run examples:
:white_check_mark: [conformance-aks-v1.12.yaml](https://github.com/cilium/cilium/actions/runs/5313019853)
:white_check_mark: [conformance-aks-v1.13.yaml](https://github.com/cilium/cilium/actions/runs/5313019842)
:white_check_mark: [conformance-aks.yaml](https://github.com/cilium/cilium/actions/runs/5313019860)

Schedule run simulated:
:white_check_mark: [conformance-aks-v1.12.yaml](https://github.com/cilium/cilium/actions/runs/5313506446)
:x: [conformance-aks-v1.13.yaml](https://github.com/cilium/cilium/actions/runs/5314131467) ,  failures are not related cluster installation
:white_check_mark: [conformance-aks.yaml](https://github.com/cilium/cilium/actions/runs/5313506456)

### CODE REVIEW

File: .github/workflows/lint-workflows.yaml [hunk 1]
```diff
@@ -157,3 +157,30 @@ jobs:
...
+        run: |
+          for dir in aws azure gke;do
+            for name in k8s-versions test-config; do
+              file_base="src/github.com/cilium/cilium/.github/actions/${dir}/${name}"
+              if [ -f ${file_base}.yaml ];then
+                yamale -s ${file_base}-schema.yaml ${file_base}.yaml;
+              fi
+            done
+          done
+
```
Comment (nbusseneau) [inline .github/workflows/lint-workflows.yaml]: Ideally this would be handled in a separate PR on which all of #26356, #26351 and #26354 depend, but it's OK to just have the other PRs depend on this one as well. I think at the very least we should split this in a separate commit though.

**Review by nbusseneau — CHANGES_REQUESTED** [review]

File: .github/workflows/conformance-aks-v1.11.yaml [hunk 2]
```diff
@@ -114,11 +112,103 @@ jobs:
             src:
               - '!(test|Documentation)/**'
 
+  setup-report:
+    runs-on: ubuntu-latest
```
Comment (nickolaev) [inline .github/workflows/conformance-aks-v1.11.yaml:116]: Do we always use latest? Shall we align to LTS (22.04)?

Comment (brlbil) [reply .github/workflows/conformance-aks-v1.11.yaml:116]: Yes, all CI uses this

File: .github/workflows/lint-workflows.yaml (hunk 1 above)
Comment (brlbil) [reply .github/workflows/lint-workflows.yaml]: It was good that this part was highlighted because tests did not cover some of the files.
Fixed it, and separated it as a commit.

File: .github/actions/azure/k8s-versions-schema.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+include: list(include('includeItem'))
```
Comment (bimmlerd) [inline .github/actions/azure/k8s-versions-schema.yaml:1]: somewhat meta, but should `.github/actions/` (maybe even all of `.github`) be CODEOWNED by by the `cilium/ci-structure` team instead of contributing? This feels far in the GHA action territory.

**Review by bimmlerd — APPROVED** [review]
I don't feel particularly qualified, see comment, but what is owned by contributing LGTM.

File: .github/actions/azure/k8s-versions.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# List of k8s version for AKS tests
+---
+include:
+  - version: "1.24"
```
Comment (tgraf) [inline .github/actions/azure/k8s-versions.yaml:4]: Are we unintentionally removing 1.23 test coverage here? The commit message says that this adds "all supported versions" so I'm not sure if the removal of 1.23 is intentional.

**Review by tgraf — CHANGES_REQUESTED** [review]

File: .github/actions/azure/k8s-versions.yaml (hunk 4 above)
Comment (nbusseneau) [reply .github/actions/azure/k8s-versions.yaml:4]: 1.23 is not supported on Azure anymore, so we removed it as part of our usual cycling (otherwise what happens the installation will fail consistently due the K8s version being refused by the Azure API on cluster creation) => https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli

**Review by nbusseneau — APPROVED** [review]

File: .github/actions/azure/k8s-versions.yaml (hunk 4 above)
Comment (tgraf) [reply .github/actions/azure/k8s-versions.yaml:4]: OK, thanks. Not clear from the commit message.

**Review by tgraf — APPROVED** [review]

File: .github/actions/azure/k8s-versions.yaml (hunk 4 above)
Comment (brlbil) [reply .github/actions/azure/k8s-versions.yaml:4]: This table was created from the supported k8s for AKS, https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar
